src/uct/rocm/rocm_cma_iface.c

```diff
@@ -0,0 +1,188 @@
+                                             uct_iface_attr_t *iface_attr)
+{
+    memset(iface_attr, 0, sizeof(uct_iface_attr_t));
+    ucs_trace("uct_rocm_cma_iface_query");
+
+    /* default values for all shared memory transports */
+    iface_attr->cap.put.min_zcopy       = 0;
+    iface_attr->cap.put.max_zcopy       = SIZE_MAX;
+    iface_attr->cap.put.opt_zcopy_align = sizeof(uint32_t);
+    iface_attr->cap.put.align_mtu       = iface_attr->cap.put.opt_zcopy_align;
+    iface_attr->cap.put.max_iov         = uct_sm_get_max_iov();
+
+    iface_attr->cap.get.min_zcopy       = 0;
+    iface_attr->cap.get.max_zcopy       = SIZE_MAX;
+    iface_attr->cap.get.opt_zcopy_align = sizeof(uint32_t);
+    iface_attr->cap.get.align_mtu       = iface_attr->cap.get.opt_zcopy_align;
+    /* Note: There is bug in ROCm 1.5 stack preventing using multiple ranges.
+            Temporally force it to 1 till ROCm 1.6 will be published */
+    iface_attr->cap.get.max_iov         = 1;//uct_sm_get_max_iov();
+
+    iface_attr->cap.am.max_iov          = 1;
+    iface_attr->cap.am.opt_zcopy_align  = 1;
+    iface_attr->cap.am.align_mtu        = iface_attr->cap.am.opt_zcopy_align;
+
+    iface_attr->iface_addr_len          = sizeof(pid_t);
+    iface_attr->device_addr_len         = UCT_SM_IFACE_DEVICE_ADDR_LEN;
+    iface_attr->ep_addr_len             = 0;
+    iface_attr->cap.flags               = UCT_IFACE_FLAG_GET_ZCOPY |
+                                          UCT_IFACE_FLAG_PUT_ZCOPY |
+                                          UCT_IFACE_FLAG_PENDING   |
+                                          UCT_IFACE_FLAG_CONNECT_TO_IFACE;
+
+    /**
+     * @todo: Add logic to query ROCr about latency information.
+     *
+     * Question: How to handle/choose the correct one in the multi-GPUs case
+     *           when latency could depends on source and target locations as
+     *           well as on which GPU will be used?
+     *
+     * Keep value the same as for CMA transport
+     *
+     */
+    iface_attr->latency.overhead        = 80e-9; /* 80 ns */
+    iface_attr->latency.growth          = 0;
+    iface_attr->bandwidth               = 10240 * 1024.0 * 1024.0; /* 10240 MB*/
+    iface_attr->overhead                = 0.4e-6; /* 0.4 us */
+
+
+    return UCS_OK;
+}
+
+static UCS_CLASS_DECLARE_DELETE_FUNC(uct_rocm_cma_iface_t, uct_iface_t);
+
+static uct_iface_ops_t uct_rocm_cma_iface_ops = {
+    .ep_put_zcopy           = uct_rocm_cma_ep_put_zcopy,
+    .ep_get_zcopy           = uct_rocm_cma_ep_get_zcopy,
+    .ep_pending_add         = ucs_empty_function_return_busy,
+    .ep_pending_purge       = ucs_empty_function,
+    .ep_flush               = uct_base_ep_flush,
+    .ep_fence               = uct_sm_ep_fence,
+    .ep_create_connected    = UCS_CLASS_NEW_FUNC_NAME(uct_rocm_cma_ep_t),
+    .ep_destroy             = UCS_CLASS_DELETE_FUNC_NAME(uct_rocm_cma_ep_t),
+    .iface_flush            = uct_base_iface_flush,
```

Comment:
need to add stubs for iface_progress_enable / iface_progress_disable
and also indent on column